### ISSUE TITLE
Memory problem in turn-by-turn monitor when using tracker multiple times

### ISSUE THREAD

**Issue by giadarol** [xsuite/xtrack#54]
This script triggers the issue:

```python
import numpy as np

import xobjects as xo
import xline as xl
import xtrack as xt

## Generate a simple sequence
sequence = xl.Line(
    elements=[xl.Drift(length=2.),
              xl.Multipole(knl=[0, 1.], ksl=[0,0]),
              xl.Drift(length=1.),
              xl.Multipole(knl=[0, -1.], ksl=[0,0])],
    element_names=['drift_0', 'quad_0', 'drift_1', 'quad_1'])

## Choose a context
context = xo.ContextCpu()         # For CPU
# context = xo.ContextCupy()      # For CUDA GPUs
# context = xo.ContextPyopencl()  # For OpenCL GPUs

## Transfer lattice on context and compile tracking code
tracker = xt.Tracker(_context=context, sequence=sequence)

## Build particle object on context
n_part = 200
particles = xt.Particles(_context=context,
                        p0c=6500e9,
                        x=np.random.uniform(-1e-3, 1e-3, n_part),
                        px=np.random.uniform(-1e-5, 1e-5, n_part),
                        y=np.random.uniform(-2e-3, 2e-3, n_part),
                        py=np.random.uniform(-3e-5, 3e-5, n_part),
                        zeta=np.random.uniform(-1e-2, 1e-2, n_part),
                        delta=np.random.uniform(-1e-4, 1e-4, n_part),
                        )

## Track (saving turn-by-turn data)
n_turns = 100
tracker.track(particles, num_turns=n_turns,
              turn_by_turn_monitor=True)

## Track second time (saving turn-by-turn data)
tracker.track(particles, num_turns=n_turns,
              turn_by_turn_monitor=True)

## Try to access turn-by-turn data
tracker.record_last_track.x
```

On CPU I get:
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-1fe53bc9bc1a> in <module>
     43 
     44 ## Try to access turn-by-turn data
---> 45 tracker.record_last_track.x

~/Desktop/20210303_xfields_gpudev/xtrack/xtrack/monitors.py in __get__(self, container, ContainerType)
     59 
     60     def __get__(self, container, ContainerType=None):
---> 61         vv = getattr(container.data, self.name)
     62         n_cols = container.stop_at_turn - container.start_at_turn
     63         n_rows = container.n_records // n_cols

~/Desktop/20210303_xfields_gpudev/xtrack/xtrack/dress.py in __get__(self, container, ContainerType)
     14     def __get__(self, container, ContainerType=None):
     15         if self.isnplikearray:
---> 16             return getattr(container._xobject, self.name).to_nplike()
     17         elif hasattr(container, '_dressed_'+self.name):
     18             return getattr(container, '_dressed_'+self.name)

~/Desktop/20210303_xfields_gpudev/xobjects/xobjects/array.py in to_nplike(self)
    573         cshape = [shape[ii] for ii in self._order]
    574         if hasattr(self._itemtype, "_dtype"):
--> 575             arr = self._buffer.to_nplike(
    576                 self._offset + self._data_offset, self._itemtype._dtype, cshape
    577             ).transpose(self._order)

~/Desktop/20210303_xfields_gpudev/xobjects/xobjects/context_cpu.py in to_nplike(self, offset, dtype, shape)
    445         # dtype=np.dtype(dtype)
    446         # return self.buffer[offset:].view(dtype)[:count].reshape(*shape)
--> 447         return np.frombuffer(
    448             self.buffer, dtype=dtype, count=count, offset=offset
    449         ).reshape(*shape)

ValueError: buffer is smaller than requested size
```

**Comment by giadarol** [comment]
A somewhat related issue has been reported by @phermes, with memory getting corrupted after many repeated tracker calls with turn-by-turn monitor on. Examples provided by @phermes should be rechecked when this issue is fixed, to see if his issue is indeed related.

**Comment by giadarol** [comment]
One problem found: ```_capacity``` non correctly initialized. To be fixed.

**Comment by giadarol** [comment]
Solved by PR #55
